src/pages/task/TestsTable.tsx

```diff
@@ -1,68 +1,54 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
+import { TestsTableCore } from "pages/task/TestsTableCore";
 import { useLocation, useHistory } from "react-router-dom";
+import {
+  Categories,
+  RequiredQueryParams,
+  Sort,
+  ValidInitialQueryParams,
+  Limit
+} from "pages/task/types";
 import queryString from "query-string";
 
-enum RequiredQueryParams {
-  Sort = "sort",
-  Category = "category",
-  Page = "page",
-  Limit = "limit"
-}
-
-enum Categories {
-  TestName = "TEST_NAME",
-  Duration = "DURATION",
-  Status = "STATUS"
-}
-
-enum Sort {
-  Desc = "-1",
-  Asc = "1"
-}
-
 enum DefaultQueryParams {
   Sort = "1",
   Category = "TEST_NAME",
   Page = "0",
   Limit = "0"
 }
 
-export const TestsTable: React.FC = () => {
+export const TestsTable: React.FC<Limit> = ({ limit }) => {
   const { pathname, search } = useLocation();
   const { replace } = useHistory();
+  const [validInitialQueryParams, setValidInitialQueryParams] = useState<
+    ValidInitialQueryParams
+  >();
 
   useEffect(() => {
     const parsed = queryString.parse(search);
-    const category = (
-      JSON.stringify(parsed[RequiredQueryParams.Category]) || ""
-    ).toUpperCase();
-    const limitNum = parseInt(
-      JSON.stringify(parsed[RequiredQueryParams.Limit]),
-      10
-    );
-    const pageNum = parseInt(
-      JSON.stringify(parsed[RequiredQueryParams.Page]),
-      10
-    );
+    const category = (parsed[RequiredQueryParams.Category] || "")
+      .toString()
+      .toUpperCase();
     const sort = parsed[RequiredQueryParams.Sort];
     if (
       (sort !== Sort.Desc && sort !== Sort.Asc) ||
       (category !== Categories.TestName &&
         category !== Categories.Duration &&
-        category !== Categories.Status) ||
-      !Number.isInteger(pageNum) ||
-      pageNum < 0 ||
-      !Number.isInteger(limitNum) ||
-      limitNum < 0
+        category !== Categories.Status)
     ) {
       parsed[RequiredQueryParams.Category] = DefaultQueryParams.Category;
       parsed[RequiredQueryParams.Sort] = DefaultQueryParams.Sort;
-      parsed[RequiredQueryParams.Page] = DefaultQueryParams.Page;
-      parsed[RequiredQueryParams.Limit] = DefaultQueryParams.Limit;
       const nextQueryParams = queryString.stringify(parsed);
       replace(`${pathname}?${nextQueryParams}`);
+    } else if (!validInitialQueryParams) {
+      setValidInitialQueryParams({
+        initialCategory: parsed[RequiredQueryParams.Category],
+        initialSort: parsed[RequiredQueryParams.Sort]
```

Comment:
this should probably be what the default value of `validInitialQueryParams` is set to. then if they're not valid they can fall back to the defaults.

Reply:
I actually intended the initial query params to be empty. In the current implementation, if the query params read from the URL are invalid, we replace the values in the URL , otherwise, we set `validInitialQueryParams` to what was in the the URL. The table won't render until the query params read from the URL are valid. I could provide default props if I had a third boolean useState that represents when the URL has valid query params. 
The reason I put the initial query params in useState to begin with is because I `useQuery` needs them as props for the initial fetch and also because the antd table needs to know to know the sort category and direction for its initial render.